functions/controllers/message/readMessages.js

```diff
@@ -0,0 +1,88 @@
+const HttpStatus = require('http-status-codes');
+const { status, message } = require('../../util/constants');
+
+const { error, success } = status;
+const { somethingWentWrong, chatSessionDoesnotExist } = message;
+const { INTERNAL_SERVER_ERROR, BAD_REQUEST, OK } = HttpStatus;
+
+const errors = {
+    error2: res => {
+        res.status(INTERNAL_SERVER_ERROR).json({
+            message: somethingWentWrong,
+            status: error,
+        });
+    },
+    error3: res => {
+        res.status(BAD_REQUEST).json({
+            message: chatSessionDoesnotExist,
+            status: error,
+        });
+    },
+};
+
+const successMessage = {
+    success1: (res, chats) => {
+        res.status(OK).json({
+            message: chats,
+            status: success,
+        });
+    },
+};
+
+const getMessages = async (req, res, db) => {
+    const chats = [];
+    try {
+        const messages = await db
+            .collection('messages')
+            .where('roomId', '==', req.params.roomId)
+            .get();
+        messages.docs.forEach(async mes => {
+            chats.push(mes.data());
+        });
```

Comment:
Why are you using `async` here?
This is more neat and optimal
```
const chats = messages.docs.map(mes =>mes.data());
```

Reply:
I have to use for each so I can work on the array.. And I did not want the flow to skip that before execution

Reply:
But you are only pushing docs data to the array chats.